src/script/lua_api/l_noise.cpp

```diff
@@ -499,6 +511,51 @@ int LuaPcgRandom::l_rand_normal_dist(lua_State *L)
 	return 1;
 }
 
+int LuaPcgRandom::l_get_state(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	LuaPcgRandom *o = checkObject<LuaPcgRandom>(L, 1);
+
+	u64 state, inc;
+	o->m_rnd.getState(state, inc);
+
+	std::stringstream s_state;
+	
+	s_state << std::hex << std::setw(64) << std::setfill('0') << state;
+	s_state << std::hex << std::setw(64) << std::setfill('0') << inc;
```

Comment:
I am pretty sure you don't need to set those modifiers again

Reply:
I have tried it without setting them again. But it does not work as I expected. So, I have been forced to set those modifiers two times.

Reply:
At first I have used this `s_state << std::hex << std::setw(64) << std::setfill('0') << state << inc;`

But the result string does not contain 128 characters, `l_set_state` function said.